DQMOffline/Muon/src/DiMuonHistograms.cc

```diff
@@ -180,29 +175,30 @@ void DiMuonHistograms::analyze(const edm::Event & iEvent,const edm::EventSetup&
 	  }
 	}
 	// Also Tight-Tight Muon Selection
-
-	if ( muon::isTightMuon(*recoMu1, vtx)  && 
-	     muon::isTightMuon(*recoMu2, vtx) ) { 
-  	  
+	if ( muon::isTightMuon(*muon1, vtx)  && 
+	     muon::isTightMuon(*muon2, vtx) ) { 
+	  test->Fill(InvMass);
 	  LogTrace(metname)<<"[DiMuonHistograms] Tight-Tight pair"<<endl;
-	  for (unsigned int iEtaRegion=0; iEtaRegion<3; iEtaRegion++){
-	    if (iEtaRegion==0) {EtaCutMin= 0.;         EtaCutMax=2.4;       }
-	    if (iEtaRegion==1) {EtaCutMin= etaBMin;    EtaCutMax=etaBMax;   } 
-	    if (iEtaRegion==2) {EtaCutMin= etaECMin;   EtaCutMax=etaECMax;  }
-
-	    if(fabs(recoCombinedGlbTrack1->eta())>EtaCutMin && fabs(recoCombinedGlbTrack1->eta())<EtaCutMax && 
-	       fabs(recoCombinedGlbTrack2->eta())>EtaCutMin && fabs(recoCombinedGlbTrack2->eta())<EtaCutMax){
-	      if (InvMass > 55. && InvMass < 125.) TightTightMuon[iEtaRegion]->Fill(InvMass);
+	  if (charge < 0){
+	    for (unsigned int iEtaRegion=0; iEtaRegion<3; iEtaRegion++){
+	      if (iEtaRegion==0) {EtaCutMin= 0.;         EtaCutMax=2.4;       }
+	      if (iEtaRegion==1) {EtaCutMin= etaBMin;    EtaCutMax=etaBMax;   } 
+	      if (iEtaRegion==2) {EtaCutMin= etaECMin;   EtaCutMax=etaECMax;  }
```

Comment:
could these if statements go into a place where they get executed once? (e.g., just store them into an array that you then loop over)
@sscruz